vm/actor/tests/deadline_test.rs

```diff
@@ -0,0 +1,671 @@
+        (PART_SIZE * 4 - 1) as u64,
+        (PART_SIZE * 6) as u64,
+        (PART_SIZE * 8 - 1) as u64,
+        (PART_SIZE * 9) as u64,
+    ]);
+
+    let (first_idx, sector_count) = partitions_for_deadline(&mut dl, PART_SIZE, 0).unwrap();
+    assert_eq!(0, first_idx);
+    assert_eq!(PART_SIZE, sector_count as usize);
+
+    let (second_idx, second_count) = partitions_for_deadline(&mut dl, PART_SIZE, 1).unwrap();
+    assert_eq!(1, second_idx);
+    assert_eq!(PART_SIZE * 2, second_count as usize);
+
+    let (third_idx, third_count) = partitions_for_deadline(&mut dl, PART_SIZE, 2).unwrap();
+    assert_eq!(3, third_idx);
+    assert_eq!(PART_SIZE * 4 - 1, third_count as usize);
+
+    let (fourth_idx, fourth_count) = partitions_for_deadline(&mut dl, PART_SIZE, 3).unwrap();
+    assert_eq!(7, fourth_idx);
+    assert_eq!(PART_SIZE * 6, fourth_count as usize);
+
+    let (fifth_idx, fifth_count) = partitions_for_deadline(&mut dl, PART_SIZE, 4).unwrap();
+    assert_eq!(13, fifth_idx);
+    assert_eq!(PART_SIZE * 8 - 1, fifth_count as usize);
+
+    let (sixth_idx, sixth_count) = partitions_for_deadline(&mut dl, PART_SIZE, 5).unwrap();
+    assert_eq!(21, sixth_idx);
+    assert_eq!(PART_SIZE * 9, sixth_count as usize);
+
+    let (third_idx, third_count) =
+        partitions_for_deadline(&mut dl, PART_SIZE, WPOST_PERIOD_DEADLINES - 1).unwrap();
+    assert_eq!(30, third_idx as usize);
+    assert_eq!(0, third_count as usize);
+}
+#[test]
+#[should_panic(expected = r#"invalid partition 0 at deadline 0 with first 0, count 0"#)]
+fn zero_partitions_at_empty_deadline_test() {
+    let mut dls = Deadlines::new();
+    dls.due[1] = bf_seq(0, 1);
+
+    // No partitions at deadline 0
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 0, &[0]).unwrap();
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 2, &[0]).unwrap();
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 2, &[1]).unwrap();
+    compute_partitions_sector(&mut dls, PART_SIZE as u64, 2, &[2]).unwrap();
+}
+
+#[test]
+fn single_sector_partition_test() {
+    let mut dls = Deadlines::new();
+    dls.due[1] = bf_seq(0, 1);
+    let partitions = compute_partitions_sector(&mut dls, PART_SIZE as u64, 1, &[0]).unwrap();
+    assert_eq!(1, partitions.clone().len());
+
+    assert_bf_equal(bf_seq(0, 1), partitions.get(0).unwrap().clone());
+}
+
+#[test]
+fn full_partition_test() {
+    let mut dls = Deadlines::new();
+    dls.due[10] = bf_seq(1234, PART_SIZE);
+
+    let partitions = compute_partitions_sector(&mut dls, PART_SIZE as u64, 10, &[0]).unwrap();
+    assert_eq!(1, partitions.clone().len());
+
+    assert_bf_equal(bf_seq(1234, PART_SIZE), partitions.get(0).unwrap().clone());
+}
+
+#[test]
+fn full_plus_partial_test() {
+    let mut dls = Deadlines::new();
+    dls.due[10] = bf_seq(5555, PART_SIZE + 1);
+
+    let mut partitions = compute_partitions_sector(&mut dls, PART_SIZE as u64, 10, &[0]).unwrap();
+    assert_eq!(1, partitions.clone().len());
```

Comment:
can you just remove all clones for checking length? never need to clone to call a function that doesn't require a move